### PR TITLE
Spark 3.0 / Scala 2.12 compatible build

### PR DESCRIPTION
**Pull request by WeichenXu123**
Spark 3.0 / Scala 2.12 compatible build.

Take over https://github.com/combust/mleap/pull/661

* Add numerical tolerance in comparing float/double in test.
* Restore some code change in ScalerModel
* fixed the kryo dependency (exclude "com.esotericsoftware.kryo" % "kryo")
* merge master
* revert commit 

Test against spark-3.0.0-rc1

### CODE REVIEW

**Comment by lucagiovagnoli** [comment]
Can I ask why opening a third PR?  The context in all the comments in https://github.com/combust/mleap/pull/665 was useful. I'm also wondering if there's a way to reduce the number of changes in this PR as much as possible (and number of commits) to make it easier to test and review

**Comment by lucagiovagnoli** [comment]
There's a lot of changes from a branch of mine, I think the code will be more readable if you rebase and squash the commits. Also, I wonder: are all the other code  changes necessary for spark 3.0 ? If you introduced code fixes unrelated to the bump, It'd be great to have them in separate PR/issues

**Comment by mengxr** [comment]
@lucagiovagnoli We need to create a new PR to re-target at spark-3.0.0 branch, per discussion at https://github.com/combust/mleap/pull/671#issuecomment-625383448

@WeichenXu123 Could you do the following:

1. Submit one PR to add Spark test dependencies and change parity tooling to allow numerical tolerance? Submit it to master. I think the same code should work for both 2.4 and 3.0.
2. Then we reset spark-3.0 branch to master.
3. Submit necessary changes for Spark 3.0. Exclude XGBoost from the compile/build/test, instead of removing its source code.
4. Fix XGBoost issues once they have a Spark 3.0 compatible release.

@lucagiovagnoli The PR in 3) would still contain large changes, which are all necessary to be compatible with Spark 3.0. But I don't think squash commits is necessary for review. We already went multiple review rounds on the changes. When we merge, we can use squad merge to avoid messy commits in the git history.

**Comment by WeichenXu123** [comment]
@mengxr
> Submit one PR to add Spark test dependencies and change parity tooling to allow numerical tolerance

Create PR here. https://github.com/combust/mleap/pull/680

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,14 @@ scala:
 jdk:
   - oraclejdk8  # sbt +compile fails on jdk11 (xenial's default)
 
+addons:
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
```
Comment (mengxr) [inline .travis.yml:17]: Is it needed?

File: project/Common.scala [hunk 2]
```diff
@@ -28,7 +28,7 @@ object Common {
       if(isSnapshot.value) {
         Seq(
           "Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots",
-          "ASF Snapshots" at "https://repository.apache.org/content/groups/snapshots"
+          "ASF spark" at "https://repository.apache.org/content/repositories/orgapachespark-1341"
```
Comment (mengxr) [inline project/Common.scala:31]: What is this repo? Could you add an inline comment?

File: python/tox.ini [hunk 3]
```diff
@@ -6,10 +6,12 @@ skipdist = true
 passenv = SCALA_CLASS_PATH
 deps =
         -rrequirements-dev.txt
+        nose
+        coverage
 commands =
         nosetests --with-coverage \
                 --cover-package=mleap \
-                --nologcapture -s \
```
Comment (mengxr) [inline python/tox.ini:12]: Could you remind me why removing "-s"?

File: travis/travis.sh [hunk 4]
```diff
@@ -3,7 +3,7 @@
 if [[ $TRAVIS_BRANCH == 'master' ]] && [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
   source travis/extract.sh
   source travis/docker.sh
-  sbt "+ test" "+ mleap-executor-tests/test" "+ mleap-benchmark/test" "mleap-xgboost-runtime/test" "mleap-xgboost-spark/test" "+ publishSigned" "mleap-xgboost-runtime/publishSigned" "mleap-xgboost-spark/publishSigned" "mleap-serving/docker:publish" "mleap-spring-boot/docker:publish"
```
Comment (mengxr) [inline travis/travis.sh:6]: Could you create a GitHub issue to make xgboost test work and link it here? So we can properly track the follow-up work.

**Comment by WeichenXu123** [comment]
Xgboost follow-up ticket https://github.com/combust/mleap/issues/683

File: mleap-spark/src/test/scala/org/apache/spark/ml/parity/feature/StopWordsRemoverParitySpec.scala [hunk 5]
```diff
@@ -1,5 +1,7 @@
 package org.apache.spark.ml.parity.feature
 
+import java.util.Locale
```
Comment (mengxr) [inline mleap-spark/src/test/scala/org/apache/spark/ml/parity/feature/StopWordsRemoverParitySpec.scala:3]: minor: not used?

**Pull request merged**